quotes-react-app/src/Main/Main.jsx

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from "react";
+
+export const Main = () => {
+  const [quotes, setQuotes] = useState([]);
+  const [randomQuote, setRandomQuote] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+
+  const quotesApi = "https://quotes-server-kxpndusosfe0.runkit.sh/quotes";
+
+  useEffect(() => {
+    const fetchQuotes = async (quotesSource) => {
+      try {
+        const response = await fetch(quotesSource);
+        const data = await response.json();
+        setQuotes(data);
+        setIsLoading(false);
+        setRandomQuote(pickRandomQuote(data));
+      } catch (error) {
+        console.error("Error fetching quotes:", error);
+      }
+    };
+
+    fetchQuotes(quotesApi);
+  }, []);
```

Comment:
The `fetchQuotes` function does not need to be created, the first parameter of the useEffect hook is a function, so it could be simplified to the following:

```jsx
useEffect(async () => {...}, []);
```

Also, some extra reading, have you seen the [useMemo()](https://react.dev/reference/react/useMemo) hook before?

Reply:
Just spotted a potential issue here.

You are setting the `isLoading` state to false before setting `randomQuote`, so the code on line 38 will be returned after the `isLoading` state changes, however since the `randomQuote` has not been set there will likely be an exception thrown from the code on lines 41 and 42.

This is because when the `loading` state changes, `randomQuote` will still equal the default value set on line 5 which is an empty object, so when you try and access the `quote` and `author` properties on line 41/42, they will be undefined.

Reply:
Thank you so much.
So can I just call setRandomQuote before setIsLoading?